src/prometheus-pushgateway/config/prometheus-pushgateway.yaml

```diff
@@ -1,21 +1,10 @@
-# Copyright (c) Microsoft Corporation
-# All rights reserved.
-#
-# MIT License
-#
-# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
-# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
-# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
-# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
-# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
-#
-# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
-# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
-# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
-# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
 
 service_type: "common"
 
 port: 9097
-scrape_interval: 30
+scrape_interval: 30 # seconds
+metrics-cleaner:
+  schedule: '0 0 * * *'
+  clean-period: 7 # days
```

Comment:
For staled metrics. If it doesn't be deleted form push gateway, will it be reported to prometheus? I mean if a job finished at 10:00 am, we run clean daemon at 10:00 pm. Will we still see the job metrics between 10:00 am to 10:00 pm

Reply:
> For staled metrics. If it doesn't be deleted form push gateway, will it be reported to prometheus? I mean if a job finished at 10:00 am, we run clean daemon at 10:00 pm. Will we still see the job metrics between 10:00 am to 10:00 pm

Yes